src/datasets/packaged_modules/parquet/parquet.py

```diff
@@ -63,7 +64,7 @@ def _cast_table(self, pa_table: pa.Table) -> pa.Table:
 
     def _generate_tables(self, files):
         schema = self.info.features.arrow_schema if self.info.features is not None else None
-        if self.info.features is not None and self.config.columns is not None:
```

Comment:
We want to run this check only if the features/schema is explicitly provided in `load_dataset` (`self.config.features is not None`). Otherwise (features from YAML or Parquet metadata), we can assume the features are aligned with the Parquet data.

Reply:
What if columns are provided that are not present in the Parquet? Or is the error from the Parquet reader simply propagated?

Reply:
Yes, we propagate the Parquet reader's error is that scenario.